onellm/files.py

```
@@ -49,24 +55,136 @@ def upload(
+                    except (OSError, IOError):
+                        # File is not seekable, will need to read
+                        pass
+                
+                # If we got the size, validate it
+                if file_size is not None:
+                    max_mb = max_size / (1024 * 1024)
+                    actual_mb = file_size / (1024 * 1024)
+                    if file_size > max_size:
+                        raise InvalidRequestError(
+                            f"File too large: {actual_mb:.2f}MB exceeds {max_mb:.2f}MB"
+                        )
```

Ruff (0.13.3)

Avoid specifying long messages outside the exception class

(TRY003)